flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/e2e/resources/KafkaContainerizedExternalSystem.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kafka.source.e2e.resources;
+
+import org.apache.flink.connectors.test.common.external.ContainerizedExternalSystem;
+import org.apache.flink.connectors.test.common.utils.FlinkContainers;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.util.Collections;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Kafka external system based on {@link KafkaContainer} from <a
+ * href="https://www.testcontainers.org/">Testcontainers</a>.
+ *
+ * <p>This external system is also integrated with a Kafka admin client for topic management.
+ */
+public class KafkaContainerizedExternalSystem
+        implements ContainerizedExternalSystem<KafkaContainerizedExternalSystem> {
```

Comment:
I'd convert `ContainerizedExternalSystem` into a abstract base class where the docker specific stuff and life-cycle is already implemented.